python/test/gherkin_test.py

```diff
@@ -89,3 +91,202 @@ def test_change_the_default_language():
     }
 
     assert expected == feature_file
+
+def test_parsing_markdown_does_not_parse_a_feature_description():
+    parser = Parser()
+    matcher = GherkinInMarkdownTokenMatcher()
+    
+    feature_file = """# Feature: hello
+This is the
+description
+"""
+    ast = parser.parse(TokenScanner(feature_file), matcher)
+    expected = {
+        'feature': {
+          'tags': [],
+          'description': '',
+          'location': { 'line': 1, 'column': 3 },
+          'language': 'en',
+          'keyword': 'Feature',
+          'name': 'hello',
+          'children': [],
+        },
+        'comments': [],
+      }
+    assert ast == expected
+
+def test_parsing_markdown_parses_a_feature_without_a_hash_Feature_header():
+    parser = Parser()
+    matcher = GherkinInMarkdownTokenMatcher()
+    feature_file = """# Hello
+This is the
+description
+
+## Scenario: hello
++ Given a step
+
+## Some other header
+"""
+    ast = parser.parse(TokenScanner(feature_file), matcher)
+    expected = {
+        'feature': {
+          'tags': [],
+          'location': {
+            'line': 1,
+            'column': 1,
+          },
+          'language': 'en',
+          'name': '# Hello',
+          'description': '',
+          'children': [
+            {
+              'scenario': {
+                'id': '1',
+                'tags': [],
+                'location': {
+                  'line': 5,
+                  'column': 4,
+                },
+                'keyword': 'Scenario',
+                'name': 'hello',
+                'description': '',
+                'steps': [
+                  {
+                    'id': '0',
+                    'location': {
+                      'line': 6,
+                      'column': 3,
+                    },
+                    'keyword': 'Given ',
+                    'keywordType': "Context",
+                    'text': 'a step',
+                  },
+                ],
+                'examples': [],
+              },
+            },
+          ],
+        },
+        'comments': [],
+      }
+    assert ast == expected
+
+def test_it_parses_markdown_data_tables_with_headers():
+  parser = Parser()
+  matcher = GherkinInMarkdownTokenMatcher()
+  markdown = """## Feature: DataTables
+
+### Scenario: minimalistic
+
+* Given a simple data table 
+  | foo | bar |
+  | --- | --- |
+  | boz | boo |
+"""
+  ast = parser.parse(TokenScanner(markdown), matcher)
+  expected = {
+        "comments": [
+          {
+            "location": {
+              "column": 3,
+              "line": 7
+            },
+            "text": None
```

Comment:
Note discrepancy with Javascript version and causes issues when executing acceptance tests.

To resolve:
- Either add the text of the table header line OR
- Remove the entry entirely - I don't know how to do this in the Python implementation (would need pointing in the right direction).